### PR TITLE
Add functionality to draw the grid

### PR DESCRIPTION
**Pull request by Hossam-Tarek**
- Add A function to draw the grid using a callback function.
- Enhance the docs to look nicer and also show more details and how to use the class.

### CODE REVIEW

File: js/Grid.js [hunk 1]
```diff
@@ -127,11 +127,11 @@ export default class Grid {
         return count;
     }
 
-    // Draws the whole grid by drawing each element using a callback function.
-    drawGrid(callback) {
-        for (let i = 0; i < this.#rows; i++) {
-            for (let j = 0; j < this.#columns; j++) {
-                callback(this.#grid[i][j]);
+    // Executes a provided function once for each grid element.
```
Comment (ahmed-fawzy-metwally) [inline js/Grid.js:130]: Great job :)
but may update the comment to be The `forEach()` method executes a provided function once for each grid element.
To make it clear that this function itself does nothing but its parameter do all thing.

File: js/Grid.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ This class is meant to deal with the memory game grid throw simple functions, wi
 | [getColumns](#get-columns)() | Number | Returns the number of columns in the grid.
 | [getRows](#get-rows)() | Number | Returns the number of rows in the grid.
 | [getElement](#get-element)(row, column) | String | Returns the specified element by row and column in the grid.
-| [drawGrid](#draw-grid)(callback) | undefined | Draws the whole grid by drawing each element using a callback function.
+| [forEach](#for-each)(callback) | undefined | Executes a provided function once for each grid element.
```
Comment (ahmed-fawzy-metwally) [inline js/Grid.md:11]: Greate documentation for `Grid` class.
Remember to edit the description of `forEach` method As I mentioned before

**Review by ahmed-fawzy-metwally — CHANGES_REQUESTED** [review]

File: js/Grid.js (hunk 1 above)
Comment (Hossam-Tarek) [reply js/Grid.js:130]: That's what I'm typing above but in other words.

File: js/Grid.md (hunk 2 above)
Comment (Hossam-Tarek) [reply js/Grid.md:11]: I think it's clear enough no need for additional words every devloper knows what `forEach` function do.

**Review by ahmed-fawzy-metwally — APPROVED** [review]

**Pull request merged**